sycl/doc/extensions/experimental/SYCL_EXT_ONEAPI_GRAPH.asciidoc

```diff
@@ -0,0 +1,290 @@
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_GRAPH` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+Table 1. Values of the `SYCL_EXT_ONEAPI_GRAPH` macro.
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version. Base features are supported.
+|===
+
+== SYCL Graph Terminology
+
+Table 2. Terminology.
+|===
+|Concept|Description
+|graph| Class that stores structured work units and their dependencies
+|node| The unit of work. Can have different attributes.
+|edge| Dependency between work units. Happens before relation.
+|===
+
+== Node
+
+Node is a class that can encapsulate SYCL kernel functions or host tasks for deferred execution.
+A graph has to be created first, the structure of a graph is defined second by adding nodes and edges.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+	class node{
+	};
+}
+----
+
+NOTE:
+
+== Edge
+
+A dependency between two nodes representing a happens before relationship.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+	// Adding dependency between two nodes.
+    	void make_edge(node sender, node receiver);
+}
+----
+
+== Graph
+
+Graph is a class that represents a directed acyclic graph of nodes. 
+A graph can be nested, can have multiple root nodes that are scheduled for execution first and multiple leaf nodes that are scheduled for execution last.
+Member functions as listed in Table 2 and 3 can be used to add nodes to a graph.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  class graph {
+  };
+
+}
+----
+
+=== Executable Graph
+
+`executable_graph` represents a user generated device and context specific execution object that can be submitted to a queue for execution. 
+The structure of an `executable_graph` object, such as adding nodes or edges, can not be changed.
+Each `executable_graph` object can only be executed once at the same time on its assigned queue.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  class executable_graph {
+  };
+
+}
+----
+
+
+Table 3. Constructors of the `graph` class.
+|===
+|Constructor|Description
+
+|`graph()`
+|Creates a `graph` object
+
+|`graph(graph& parent)`
+|Creates a nested `graph` object
+
+|===
+
+Table 4. Member functions of the `graph` class.
+|===
+|Member function|Description
+
+|`node add_empty_node(const std::vector<node>& dep = {});`
+|This node holds no task that is scheduled for execution. It's intended use is a synchronization point inside a graph, this node can significantly reduce the number of edges ( O(n) vs. O(n^2) ) .
+
+|`template<typename T>
+    	node add_host_node(T hostTaskCallable, const std::vector<node>& dep = {});`
```

Comment:
SYCL 2020 only has one `submit` function, which accepts both host and device tasks.  Can this be simplified to `add_node`?

Reply:
Yes I made those changes. The only drawback is that it might prevent optimization opportunities if the information that a node contains a host task cannot be derived at compile time.